site/_posts/2019-08-01-r-cran-release.md

```diff
@@ -0,0 +1,87 @@
+
+```r
+library(arrow)
+df <- read_parquet("path/to/file.parquet")
+```
+
+This function, along with the other readers in the package, takes an optional `col_select` argument, inspired by the [`vroom`](https://vroom.r-lib.org/reference/vroom.html) package. This argument lets you use the ["tidyselect" helper functions](https://tidyselect.r-lib.org/reference/select_helpers.html), as you can do in `dplyr::select()`, to specify that you only want to keep certain columns. By narrowing your selection at read time, you can load a `data.frame` with less memory overhead.
+
+For example, suppose you had written the `iris` dataset to Parquet. You could read a `data.frame` with only the columns `c("Sepal.Length", "Sepal.Width")` by doing
+
+```r
+df <- read_parquet("iris.parquet", col_select = starts_with("Sepal"))
+```
+
+Just as you can read, you can write Parquet files:
+
+```r
+write_parquet(df, "path/to/different_file.parquet")
+```
+
+## Feather files
+
+This release also includes full support for the Feather file format, providing `read_feather()` and `write_feather()`. [Feather](https://github.com/wesm/feather) was one of the initial products coming out of the Arrow project, providing an efficient, common file format language-agnostic data frame storage, along with implementations in R and Python.
+
+As Arrow progressed, development of Feather moved to the [`apache/arrow`](https://github.com/apache/arrow) project, and for the last two years, the Python implementation of Feather has just been a wrapper around `pyarrow`. This meant that as Arrow progressed and bugs were fixed, the Python version of Feather got the improvements but sadly R did not.
+
+With this release, the R implementation of Feather catches up and now depends on the same underlying C++ library as the Python version does. This should result in more reliable and consistent behavior across the two languages.
+
+We encourage all R users of `feather` to switch to using `arrow::read_feather()` and `arrow::write_feather()`.
+
+Note that both Feather and Parquet are columnar data formats that allow sharing data frames across R, Pandas, and other tools. When should you use Feather and when should you use Parquet? We currently recommend Parquet for long-term storage, as well as for cases where the size on disk matters because Parquet supports various compression formats. Feather, on the other hand, may be faster to read in because it matches the in-memory format and doesn't require deserialization, and it also allows for memory mapping so that you can access data that is larger than can fit into memory. See the [Arrow project FAQ](https://arrow.apache.org/faq/) for more.
```

Comment:
I think if you say "Parquet supports various compression formats" it might bring up some canards with the R community. It's simpler to say that "Parquet is optimized to create small files and as a result can be more expensive to read locally, but it performs very well with remote storage like HDFS or Amazon S3. Feather is designed for fast local reads, particularly with solid state drives, and is not intended for use with remote storage systems. Feather files can be memory-mapped and read in Arrow format without any deserialization while Parquet files always must be decompressed and decoded."